### PR TITLE
CLN: Refactor groupby._make_wrapper

### LINKED ISSUE #48028: CLN: Refactor groupby's _make_wrapper
Currently pandas implements various groupby ops via `_make_wrapper`. There are "allow lists" in `pandas.core.groupby.base` for both SeriesGroupBy and DataFrameGroupBy, and the class decorator `pandas.core.groupby.generic.pin_allowlisted_properties` adds these as properties to the SeriesGroupBy and DataFrameGroupBy classes.

Instead, should we convert `_make_wrapper` into a normal method (e.g. `_op_via_apply`) and add methods for each op in the allow lists. An example of such an added method would be:

```
    @doc(DataFrame.skew.__doc__)
    def skew(
        self,
        axis=lib.no_default,
        skipna=True,
        level=None,
        numeric_only=None,
        **kwargs
    ):
        result = self._op_via_apply(
            'skew',
            axis=axis,
            skipna=skipna,
            level=level,
            numeric_only=numeric_only,
            **kwargs
        )
        return result
```

The advantage of the current approach is less boilerplate code (the method definitions) and a consistent API between e.g. DataFrame and DataFrameGroupBy ops. But the disadvantage is that the docs come out as properties (e.g. [skew](https://pandas.pydata.org/pandas-docs/dev/reference/api/pandas.core.groupby.DataFrameGroupBy.skew.html)) and anyone using Python's dynamic abilities gets incorrect results:

```
df = pd.DataFrame({'a': [1, 1, 2], 'b': [3, 4, 5]})
gb = df.groupby('a')

print(type(DataFrameGroupBy.sum))
print(type(DataFrameGroupBy.skew))
print(type(gb.sum))
print(type(gb.skew))
print(inspect.signature(df.groupby('a').sum))
print(inspect.signature(df.groupby('a').skew))

# <class 'function'>
# <class 'property'>
# <class 'method'>
# <class 'function'>
# (numeric_only: 'bool | lib.NoDefault' = <no_default>, min_count: 'int' = 0, engine: 'str | None' = None, engine_kwargs: 'dict[str, bool] | None' = None)
# (*args, **kwargs)
```

I also find the current implementation harder to understand / debug, but that is very much an opinion.

Finally, tests can be added to ensure the consistency of arguments between e.g. `DataFrame.skew` and `DataFrameGroupBy.skew` using Python's `inspect` module.

cc @jreback @jbrockmendel @mroeschke for any thoughts.

### PR DESCRIPTION
**Pull request by rhshadrach**
- [x] closes #48028 (Replace xxxx with the Github issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Currently keeping the docs the same, plan to make groupby-specific docstrings in the future. Note that most of the SeriesGroupBy docs don't make it into the documentation, I've opened #48399. The only change from a user-perspective here should be that Jupyter / IDEs can now get the correct signature (instead of `*args, **kwargs`).

image

image

### CODE REVIEW

File: pandas/core/groupby/groupby.py
Comment (mroeschke) [inline pandas/core/groupby/groupby.py:991]: A 1 line docstring of this function would be nice.

File: pandas/tests/groupby/test_api_consistency.py
Comment (mroeschke) [inline pandas/tests/groupby/test_api_consistency.py:19]: Side note: Doing something like this for the entire API would be really cool. It would help the ensure when adding/modifying/deprecating methods & their signatures that consistency remains.

File: pandas/core/groupby/groupby.py
Comment (rhshadrach) [reply pandas/core/groupby/groupby.py:991]: Good call - thanks. Done.

File: pandas/core/groupby/groupby.py
Comment (mroeschke) [inline pandas/core/groupby/groupby.py:998]: Just curious, what `name`s would hit this path?

**Review by mroeschke — APPROVED** [review]
Overall LGTM. Could use another set of eyes too, but happy to merge if there's no other comments.

File: pandas/core/groupby/groupby.py
Comment (rhshadrach) [reply pandas/core/groupby/groupby.py:998]: Properties - plot, is_monotonic_increasing, is_monotonic_decreasing, dtype, and dtypes.

**Comment by mroeschke** [comment]
Awesome! A lot more straightforward, thanks @rhshadrach

**Pull request merged**
